src/Goerli-DssSpell.t.sol

```diff
@@ -496,4 +497,320 @@ contract DssSpellTest is GoerliDssSpellTestBase {
+        );
+        hevm.store(
+            address(vat),
+            keccak256(abi.encode(address(end), keccak256(abi.encode(bytes32("YFI-A"), uint256(4))))),
+            bytes32(bytes32(type(uint256).max))
+        );
+        hevm.store(
+            address(vat),
+            keccak256(abi.encode(address(end), keccak256(abi.encode(bytes32("LINK-A"), uint256(4))))),
+            bytes32(bytes32(type(uint256).max))
+        );
+
+        end.cash("ETH-A", daiToRedeem);
+        end.cash("YFI-A", daiToRedeem);
+        end.cash("LINK-A", daiToRedeem);
+    }
+
+    function testCure() public {
+        assertEq(CureAbstract(cure).wards(address(end)), 0);
+
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        assertEq(CureAbstract(cure).wards(address(end)), 1);
+    }
+
+    function testFlashLegacy() public {
+        FlashAbstract flash = FlashAbstract(addr.addr("MCD_FLASH_LEGACY"));
+        assertEq(flash.max(), 500_000_000 * WAD);
+
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        uint256 vowDai = vat.dai(address(vow));
+
+        uint256 amount = 1 * MILLION;
+        assertEq(flash.toll(), 0);
+        assertEq(flash.flashFee(address(dai), amount * WAD), 0);
+        assertEq(flash.max(), 250_000_000 * WAD);
+        assertEq(flash.flashFee(address(dai), amount * WAD), 0); // 0 DAI fee on a 1M loan (on any actually)
+        flash.flashLoan(address(this), address(dai), amount * WAD, "");
+        flash.vatDaiFlashLoan(address(this), amount * RAD, "");
+        assertEq(vat.sin(address(flash)), 0);
+        assertEq(vat.dai(address(flash)), 0);
+        flash.accrue();
+        assertEq(vat.dai(address(flash)), 0);
+        assertEq(vat.dai(address(vow)), vowDai);
+    }
+
+    function testFlash() public {
+        FlashAbstract flash = FlashAbstract(addr.addr("MCD_FLASH"));
+
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        uint256 amount = 1 * MILLION;
+        assertEq(flash.flashFee(address(dai), amount * WAD), 0);
+        assertEq(flash.max(), 250_000_000 * WAD);
+        assertEq(flash.flashFee(address(dai), amount * WAD), 0); // 0 DAI fee on a 1M loan (on any actually)
+        flash.flashLoan(address(this), address(dai), amount * WAD, "");
+        flash.vatDaiFlashLoan(address(this), amount * RAD, "");
+        assertEq(vat.sin(address(flash)), 0);
+        assertEq(vat.dai(address(flash)), 0);
+    }
+
+    function onFlashLoan(
+        address initiator,
+        address token,
+        uint256 amount,
+        uint256 fee,
+        bytes calldata
+    ) external returns (bytes32) {
+        assertEq(initiator, address(this));
+        assertEq(token, address(dai));
+        assertEq(amount, 1 * MILLION * WAD);
+        assertEq(fee, 0);
+
+        dai.approve(msg.sender, 1_000_000 * WAD);
```

Comment:
mm why not `1 * MILLION * WAD` as above, or better yet just use `amount`?

Reply:
Done.